### PR TITLE
cluster up: add option to install logging components

### PR DESCRIPTION
**Pull request by csrwng**
Adds `--logging` flag to cluster up to install log aggregation components

### CODE REVIEW

**Comment by csrwng** [comment]
@jim-minter @bparees ptal
@sosiouxme @jcantrill fyi

File: pkg/bootstrap/docker/openshift/logging.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+)
+
+const (
+	loggingNamespace               = "logging"
+	svcKibana                      = "kibana-logging"
+	loggingDeployerAccountTemplate = "logging-deployer-account-template"
+	loggingDeployerTemplate        = "logging-deployer-template"
+)
+
+func instantiateTemplate(client client.Interface, mapper configcmd.Mapper, templateNamespace, templateName, targetNamespace string, params map[string]string) error {
```
Comment (bparees) [inline pkg/bootstrap/docker/openshift/logging.go]: we already have one of these for the jenkins admission controller, can we not reuse it/move it to a helper utility location?

**Comment by bparees** [comment]
I think you need @jcantrill's sign off more than mine.

**Comment by stevekuznetsov** [comment]
I had understood the executive direction of deploying logging to be an Ansible playbook -- do we _really_ want to maintain this as well?

**Comment by csrwng** [comment]
@stevekuznetsov so yes, for now ... ideally the deployer pod would do an Ansible install

File: pkg/bootstrap/docker/openshift/helper.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ const (
 
 var (
 	openShiftContainerBinds = []string{
+		"/var/log:/var/log:rw",
```
Comment (jim-minter) [inline pkg/bootstrap/docker/openshift/helper.go]: Out of interest, why is this being added?

File: pkg/bootstrap/docker/up.go [hunk 3]
```diff
@@ -720,9 +742,14 @@ func (c *ClientStartConfig) ServerInfo(out io.Writer) error {
 		metricsInfo = fmt.Sprintf("The metrics service is available at:\n"+
 			"    https://%s\n\n", openshift.MetricsHost(c.RoutingSuffix, c.ServerIP))
 	}
+	loggingInfo := ""
+	if c.ShouldInstallLogging && c.ShouldInitializeData() {
+		loggingInfo = fmt.Sprintf("The kibana logging UI is available at:\n"+
+			"    https://%s\n\n", openshift.LoggingHost(c.RoutingSuffix, c.ServerIP))
+	}
```
Comment (jim-minter) [inline pkg/bootstrap/docker/up.go]: (As much a note to self as anything): this info should probably go into oc cluster status as well - see whether this PR beats #11171 or not!

File: pkg/bootstrap/docker/openshift/helper.go (hunk 2 above)
Comment (sosiouxme) [inline pkg/bootstrap/docker/openshift/helper.go]: For aggregated logging to work, fluentd relies on the node to write symlinks in /var/log/containers that serve as metadata to the actual container logs. Without mounting /var/log the containerized node would be writing symlinks to its container filesystem where fluentd can't read them.

**Comment by smarterclayton** [comment]
This is installing the single system setup for logging to enable someone to run with it on in a semi-realistic way.

**Comment by jcantrill** [comment]
@stevekuznetsov This is the short term fix to 'oc cluster up' providing a logging stack for someone to experience logging.  In future, once ansible is availabe and packaged nicely in an image, this would need to be reworked to consume that change.

File: pkg/bootstrap/docker/openshift/logging.go (hunk 1 above)
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/logging.go]: @bparees so I looked into merging this code with the one from the admission controller, but there's very little that we can actually share. Basically we could share the call to the bulk create. However, even then, you have a custom After handler in that code that is not necessarily what I want in this case.

https://github.com/openshift/origin/blob/master/pkg/build/admission/jenkinsbootstrapper/admission.go#L112

Comment (bparees) [inline pkg/bootstrap/docker/openshift/logging.go]: ok

**Comment by jcantrill** [comment]
My only comment is regarding the template.  I don't know how much it has changed over the lifetime of logging but unfortunate it will need to be refreshed here when it changes.  May be less of an issue once we are able to rely on ansible.  Otherwise LGTM

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
[merge]

**Comment by csrwng** [comment]
https://github.com/openshift/origin/issues/11094
[merge]

**Comment by csrwng** [comment]
https://github.com/openshift/origin/issues/9203
[merge]

**Pull request merged**